test/src/test/java/jenkins/security/ApiTokenPropertyTest.java

```diff
@@ -98,4 +106,71 @@ public void security49Upgrade() throws Exception {
         u.addProperty(t);
         assertTrue(t.getApiToken().equals(Util.getDigestOf(historicalInitialValue+"somethingElse")));
     }
+
+    private void configureWebConnection(final WebClient wc, final String token) throws IOException {
+        // See https://hc.apache.org/httpcomponents-client-ga/tutorial/html/authentication.html
+        final UsernamePasswordCredentials fooCreds = new UsernamePasswordCredentials("foo", token);
+
+        URL hostUrl = j.getURL();
+        final HttpHost targetHost = new HttpHost(hostUrl.getHost(), hostUrl.getPort(), hostUrl.getProtocol());
+        CredentialsProvider credsProvider = new BasicCredentialsProvider() {
+            @Override
+            public Credentials getCredentials(AuthScope authscope) {
+                return fooCreds;
+            }
+        };
+        credsProvider.setCredentials(
+                new AuthScope("localhost", AuthScope.ANY_PORT, AuthScope.ANY_REALM),
+                fooCreds);
+
+        // Create AuthCache instance
+        AuthCache authCache = new BasicAuthCache();
+        // Generate BASIC scheme object and add it to the local auth cache
+        AuthScheme authScheme = new BasicScheme();
+        authCache.put(targetHost, authScheme);
+
+        // Add AuthCache to the execution context
+        final HttpClientContext context = HttpClientContext.create();
+        context.setCredentialsProvider(credsProvider);
+        context.setAuthCache(authCache);
+
+        wc.setCredentialsProvider(credsProvider);
+
+        // Need to create our own WebConnection that gives us control of HttpClient execution,
+        // allowing us to pass our own HttpClientContext etc. HttpWebConnection has its own
+        // private HttpClientContext instance, which means we can't authenticate properly.
```

Comment:
@jglick yeah it's horrible, but I couldn't see any other way of doing it at the time.